### PR TITLE
renorm op

### PR DESCRIPTION
**Pull request by seemingwang**
### PR types
New features

### PR changes
OPs

### Describe
renorm op

### CODE REVIEW

File: paddle/fluid/operators/renorm_op.cu [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    weight_derivative.mutable_data<T>(ctx.GetPlace());
+    RenormGradKernelFunc1<T><<<grid, block, 0, stream>>>(
+        x_data, dout_data, pow_value.mutable_data<T>(ctx.GetPlace()),
+        mul_value.mutable_data<T>(ctx.GetPlace()), numel, dimension_each, p,
+        dim_divisor);
+    std::vector<int> reduce_axis = {0, 2};
+    TensorReduceFunctorImpl<T, T, CustomSum>(pow_value, &dim_value, reduce_axis,
+                                             stream);
+    TensorReduceFunctorImpl<T, T, CustomSum>(mul_value, &weight_derivative,
+                                             reduce_axis, stream);
```
Comment (jeff41404) [inline paddle/fluid/operators/renorm_op.cu]: Recently, this [PR](https://github.com/PaddlePaddle/Paddle/pull/38135/files) add the transformop parameter in TensorReduceFunctorImpl, it's better to have the same modification.

File: paddle/fluid/operators/renorm_op.cu [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    std::vector<framework::Tensor*> outs = {&pow_value};
+    auto func = UnsignedPowFunctor<MT, T>(p);
+    const auto& cuda_ctx =
+        context.template device_context<platform::CUDADeviceContext>();
+
+    LaunchSameDimsElementwiseCudaKernel<ElementwiseType::kUnary, MT, T,
+                                        UnsignedPowFunctor<MT, T>>(
+        cuda_ctx, ins, &outs, func);
+    std::vector<int> reduce_axis = {0, 2};
+    TensorReduceFunctorImpl<T, T, CustomSum>(pow_value, &dim_value, reduce_axis,
```
Comment (jeff41404) [inline paddle/fluid/operators/renorm_op.cu]: Recently, this [PR](https://github.com/PaddlePaddle/Paddle/pull/38135/files) add the transformop parameter in TensorReduceFunctorImpl, it's better to have the same modification.

File: python/paddle/tensor/math.py [hunk 3]
```diff
@@ -1069,6 +1069,31 @@ def addmm(input, x, y, beta=1.0, alpha=1.0, name=None):
         type="addmm", inputs=inputs, attrs=attrs, outputs={"Out": out})
     return out
 
+def renorm(input, p, dim, max_norm):
```
Comment (jeff41404) [inline python/paddle/tensor/math.py]: parameter "input" should be "x" like other Python API and "dim" should be "axis"

File: paddle/fluid/operators/renorm_op.cc [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                 "int,the dimension to slice over to get the sub-tensors");
+    AddAttr<float>("max_norm", "(float, the norm upper-bound");
+    AddAttr<bool>("use_cudnn",
+                  "(bool, default false) Only used in cudnn kernel, need "
+                  "install cudnn")
+        .SetDefault(false);
+    AddAttr<bool>("use_mkldnn",
+                  "(bool, default false) Only used in mkldnn kernel")
+        .SetDefault(false);
+    AddComment(R"DOC(
```
Comment (zmxdream) [inline paddle/fluid/operators/renorm_op.cc:59]: doc 里没改

**Review by jzhang533 — DISMISSED** [review]
lgtm

**Review by TCChenlong — APPROVED** [review]

**Review by danleifeng — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]

**Review by jeff41404 — APPROVED** [review]
LGTM

**Pull request merged**